File: WinFormsApp1/UnitTestProject2/UnitTest1.vb
Using Microsoft.VisualStudio.TestTools.UnitTesting;
Using System;
Using WinFormsApp1;
Using System.Windows.Forms;


Namespace UnitTestProject1
{
    [TestClass]
    Public Class LaboratoryStaffTest
    {
        [TestMethod]
        Public void TestConnect()
        {
            WinFormsApp1 Lb = New WinFormsApp1();
            Lb.Show();
            Assert.IsTrue(Lb.WinFormsAdd1.Rows.Count > 0);
        }

        [TestMethod]
        Public void TestButtonAdd()
        {
            WinFormsAdd1 Lb = New WinFormsAdd1();
            Lb.Show();
            Try
            {
                Lb.buttonAdd.PerformClick();
            }
            Catch
            {
                Assert.Fail();
            }
        }
    }
}

File: WinFormsApp1/Test/UnitTest1.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using WinFormsApp1;
using System.Windows.Forms;



namespace Test
{
    [TestClass]
    public class Form1Test
    {
        [TestMethod]
        public void TestConnect()
        {
            WinFormsApp1 Lb = new WinFormsApp1();
            Lb.Show();
            Assert.IsTrue(Lb.dataGridView1.Rows.Count > 0);
        }

      /*  [TestMethod]
        public void TestButtonAdd()
        {
            LaboratoryStaff Lb = new LaboratoryStaff();
            Lb.Show();
            try
            {
                Lb.buttonAdd.PerformClick();
            }
            catch
            {
                Assert.Fail();*/
            }
        }
    }
}
